engine/src/main/java/org/destinationsol/game/item/ItemManager.java

```diff
@@ -231,22 +233,40 @@ public void addAllGuns(ItemContainer ic) {
         }
     }
 
-    public List<MoneyItem> moneyToItems(float amt) {
-        ArrayList<MoneyItem> res = new ArrayList<>();
-        while (amt > MoneyItem.AMT) {
-            MoneyItem example;
-            if (amt > MoneyItem.BIG_AMT) {
-                example = moneyItem(MoneyItem.BIG_AMT);
-                amt -= MoneyItem.BIG_AMT;
-            } else if (amt > MoneyItem.MED_AMT) {
-                example = moneyItem(MoneyItem.MED_AMT);
-                amt -= MoneyItem.MED_AMT;
+    /**
+     * Turn money into item drops
+     * @param amount the amount of money to turn into drops
+     */
+    public List<MoneyItem> moneyToItems(float amount) {
+        return moneyToItems(amount, -1);
+    }
+
+    /**
+     * Turn money into item drops, with a specifiable limit to the number of items
+     * @param amount the amount of money to turn into drops
+     * @param limit -1 for no limit, otherwise the maximum number of money items to create
+     */
+    public List<MoneyItem> moneyToItems(float amount, int limit) {
+        ArrayList<MoneyItem> items = new ArrayList<>();
+        int moneyAmount;
+        while (amount > MoneyItem.SMALL_AMOUNTT) {
+            if (amount > MoneyItem.HUGE_AMOUNT) {
+                moneyAmount = MoneyItem.HUGE_AMOUNT;
+            } else if (amount > MoneyItem.BIG_AMOUNT) {
+                moneyAmount = MoneyItem.BIG_AMOUNT;
+            } else if (amount > MoneyItem.MEDIUM_AMOUNT) {
+                moneyAmount = MoneyItem.MEDIUM_AMOUNT;
             } else {
```

Comment:
Maybe add `if` there too, so if you die with 15 money for example, you don't end up dropping 20?

Reply:
Thanks for noticing! Just a sec...

Reply:
I checked the code, and right now enemies and the player only drop the exact listed amount of money, so 10, 30 and 100 (before I added 1000 in this branch). the remainder is discarded, so money under 10 is not considered. I'll open an issue for it, but I think we should handle it separately from this PR